### PR TITLE
PIC 4372 db schema report

### PR DESCRIPTION
**Pull request by markreesmoj**
No changes to the implementation, so the app is still working exactly as it was - please bear this in mind when looking at the test assertion changes :-)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -95,6 +95,13 @@ task integrationTest(type: Test) {
     }
 }
 
+task initialiseDatabase(type: Test) {
+    useJUnitPlatform()
+    filter {
+        includeTestsMatching '*PingerIntTest*'
+    }
+}
+
```
Comment (markreesmoj) [inline build.gradle:104]: just run any test which initialises the ApplicationContext and therefore runs Flyway against the docker postgres instance

File: build.gradle [hunk 2]
```diff
@@ -222,9 +229,7 @@ dependencies {
 
     testAnnotationProcessor 'org.projectlombok:lombok:1.18.32'
 
-    testImplementation "org.testcontainers:postgresql:1.17.6"
     testImplementation "org.testcontainers:junit-jupiter:1.20.1"
```
Comment (markreesmoj) [inline build.gradle:232]: To be removed in a later PR :-)

File: .circleci/config.yml [hunk 3]
```diff
@@ -138,13 +138,13 @@ jobs:
...
+      name: hmpps/java_localstack_postgres
+      jdk_tag: "21.0"
+      localstack_tag: "3.0"
+      postgres_tag: "14.12"
     steps:
       - checkout
-      - run:
-          name: Upgrade gradle
-          command: |
-            ./gradlew -v
```
Comment (markreesmoj) [inline .circleci/config.yml:147]: doesn't do what it says it does, seems unnecessary

File: docker-compose.yml [hunk 4]
```diff
@@ -1,9 +1,7 @@
-version: '3.3'
-
 services:
 
   postgres:
-    image: postgres:11.15-alpine
+    image: postgres:14.12-alpine
```
Comment (markreesmoj) [inline docker-compose.yml:4]: current live version

File: docker-compose.yml [hunk 5]
```diff
@@ -1,9 +1,7 @@
-version: '3.3'
-
```
Comment (markreesmoj) [inline docker-compose.yml:2]: was outputting a warning - this is obsolete now

File: src/test/resources/application-test.yml [hunk 6]
```diff
@@ -57,10 +57,7 @@ spring:
           jwk-set-uri: ${nomis-oauth.base-url}/auth/.well-known/jwks.json
 
   datasource:
-    url: jdbc:tc:postgresql:11.1:///courtcaseservicetest?currentSchema=courtcaseservicetest&stringtype=unspecified
-    username: ituser
-    password: itpass
-
+    url: jdbc:postgresql://${database.endpoint:localhost:5432}/${database.name:postgres}?currentSchema=${database.schema.name}&user=${database.username:root}&password=${database.password:dev}&stringtype=unspecified
```
Comment (markreesmoj) [inline src/test/resources/application-test.yml:60]: only here because ssl is enabled in the `main` connection string

**Comment by markreesmoj** [comment]
If we like this, we can probably remove the `build_data_dictionary` job and associated code

**Comment by markreesmoj** [comment]
[Output when the database_schema_report job is working - atm postgresql.org is down ](https://output.circle-artifacts.com/output/job/45666d2e-63fd-45f3-9ec7-3b1c127292b6/artifacts/0/schema/report/tables/hearing.html)

File: README.md [hunk 7]
```diff
@@ -8,6 +8,10 @@ For more informations, check our [Runbook](https://dsdmoj.atlassian.net/wiki/spa
 
 ---
 
+## Testing
+
+`docker compose up localstack-court-case-service postgres`
```
Comment (Obsiye) [inline README.md:13]: I see. Remove test containers localstack and instead run localstack in a docker container externally. 🥇 awesome thank you

File: src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java [hunk 8]
```diff
@@ -270,104 +270,6 @@ private static HearingEntity getHearingEntity(HearingDefendantEntity hearingDefe
         return hearingEntity;
     }
 
-    @Test
-    // TODO: is this a possible scenario? Check with John
```
Comment (Obsiye) [inline src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java:274]: Should this be in a ticket `// TODO: is this a possible scenario? Check with John` ? as a spike mayb...

Comment (markreesmoj) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java:274]: Maybe - it is a 2 year old comment made from someone who has left to someone else who has left.
The method where the assertion is failing `hearingEntityInitService.findFirstByHearingIdFullyInitialised` is a test only method anyway

Like I say, I don't really understand what this test is for.

Comment (markreesmoj) [reply src/test/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepositoryFacadeIntTest.java:274]: An alternative would be to delete the last assertion and leave the previous one
` assertThat(hearingRepositoryFacade.save(hearingEntity)).isEqualTo(hearingEntity);`

**Review by Obsiye — APPROVED** [review]

**Pull request merged**